libs/patchwheel/parse_test.go

```diff
@@ -0,0 +1,124 @@
+package patchwheel
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCalculateNewVersion(t *testing.T) {
+	tests := []struct {
+		name             string
+		info             *WheelInfo
+		mtime            time.Time
+		expectedVersion  string
+		expectedFilename string
+	}{
+		{
+			name: "basic version",
+			info: &WheelInfo{
+				Distribution: "mypkg",
+				Version:      "1.2.3",
+				Tags:         []string{"py3", "none", "any"},
+			},
+			mtime:            time.Date(2025, 3, 4, 12, 34, 56, 780_000_000, time.UTC),
+			expectedVersion:  "1.2.3+2025030412345678",
+			expectedFilename: "mypkg-1.2.3+2025030412345678-py3-none-any.whl",
+		},
+		{
+			name: "existing plus version",
+			info: &WheelInfo{
+				Distribution: "mypkg",
+				Version:      "1.2.3+local",
+				Tags:         []string{"py3", "none", "any"},
+			},
+			mtime:            time.Date(2025, 3, 4, 12, 34, 56, 100_000_000, time.UTC),
+			expectedVersion:  "1.2.3+2025030412345610",
+			expectedFilename: "mypkg-1.2.3+2025030412345610-py3-none-any.whl",
+		},
+		{
+			name: "complex distribution name",
+			info: &WheelInfo{
+				Distribution: "my-pkg-name",
+				Version:      "1.2.3",
+				Tags:         []string{"py3", "none", "any"},
+			},
+			mtime:            time.Date(2025, 3, 4, 12, 34, 56, 0, time.UTC),
+			expectedVersion:  "1.2.3+2025030412345600",
+			expectedFilename: "my-pkg-name-1.2.3+2025030412345600-py3-none-any.whl",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			newVersion, newFilename := CalculateNewVersion(tt.info, tt.mtime)
+			if newVersion != tt.expectedVersion {
+				t.Errorf("expected version %s, got %s", tt.expectedVersion, newVersion)
+			}
+			if newFilename != tt.expectedFilename {
+				t.Errorf("expected filename %s, got %s", tt.expectedFilename, newFilename)
+			}
+		})
+	}
+}
+
+func TestParseWheelFilename(t *testing.T) {
+	tests := []struct {
+		filename         string
+		wantDistribution string
+		wantVersion      string
+		wantTags         []string
+		wantErr          bool
+	}{
+		{
+			filename:         "myproj-0.1.0-py3-none-any.whl",
+			wantDistribution: "myproj",
+			wantVersion:      "0.1.0",
+			wantTags:         []string{"py3", "none", "any"},
+			wantErr:          false,
+		},
+		{
+			filename:         "myproj-0.1.0+20240303123456-py3-none-any.whl",
```

Comment:
we need one more test case with a build tag here

Reply:
Added this and other tests from https://github.com/wheelodex/wheel-filename/blob/f5a72b560d016cc9663c8b5a094c96dc338a2209/test/test_parse.py